cmd/k3s/main.go

```diff
@@ -100,6 +101,30 @@ func findDataDir() string {
 	return dataDir
 }
 
+// findPreferBundledBin searches for prefer-bundled-bin from the CLI args and config file.
+func findPreferBundledBin(args []string) bool {
+	preferRes := configfilearg.MustFindString(args, "prefer-bundled-bin")
+	for i, arg := range args {
+		flagName := "--prefer-bundled-bin"
+		if flagName == arg {
+			if len(args) > i+1 {
+				if res, err := strconv.ParseBool(args[i+1]); err == nil {
+					return res
+				}
+			}
+			//Naked flag, assume true
+			return true
```

Comment:
I think returning here will cause odd behavior. Remember that we support merging config from a variety of sources, so this might end up passed multiple times. We could end up with something like:
`--prefer-bundled-bin --foo=bar --prefer-bundled-bin=false`

Returning here in case of a naked flag would cause it to miss parsing the second instance.